x-pack/plugin/enrich/src/test/java/org/elasticsearch/xpack/enrich/EnrichPolicyExecutionStatsTrackerTests.java

```diff
@@ -0,0 +1,100 @@
+        assertEquals(0L, stats.getTotalRepeatExecutionCount());
+        assertEquals(0L, stats.getTotalTimeBetweenExecutions());
+        assertEquals(0L, stats.getAvgTimeBetweenExecutions());
+        assertEquals(-1L, stats.getMinTimeBetweenExecutions());
+        assertEquals(0L, stats.getMaxTimeBetweenExecutions());
+    }
+
+    public void testUniqueRunStats() {
+        EnrichPolicyExecutionStatsTracker tracker = new EnrichPolicyExecutionStatsTracker();
+        tracker.captureCompletedExecution("test1", 1000, 1025);
+        tracker.captureCompletedExecution("test2", 1050, 1100);
+        EnrichStatsAction.Response.ExecutionStats stats = tracker.getStats();
+        assertNotNull(stats);
+        assertEquals(2L, stats.getTotalExecutionCount());
+        assertEquals(75L, stats.getTotalExecutionTime());
+        assertEquals(25L, stats.getMinExecutionTime());
+        assertEquals(50L, stats.getMaxExecutionTime());
+        assertEquals(0L, stats.getTotalRepeatExecutionCount());
+        assertEquals(0L, stats.getTotalTimeBetweenExecutions());
+        assertEquals(0L, stats.getAvgTimeBetweenExecutions());
+        assertEquals(-1L, stats.getMinTimeBetweenExecutions());
+        assertEquals(0L, stats.getMaxTimeBetweenExecutions());
+    }
+
+    public void testMultipleRunStats() {
+        EnrichPolicyExecutionStatsTracker tracker = new EnrichPolicyExecutionStatsTracker();
+        tracker.captureCompletedExecution("test", 1000, 1050);
+        tracker.captureCompletedExecution("test", 1100, 1150);
+        EnrichStatsAction.Response.ExecutionStats stats = tracker.getStats();
+        assertNotNull(stats);
+        assertEquals(2L, stats.getTotalExecutionCount());
+        assertEquals(100L, stats.getTotalExecutionTime());
+        assertEquals(50L, stats.getMinExecutionTime());
+        assertEquals(50L, stats.getMaxExecutionTime());
+        assertEquals(1L, stats.getTotalRepeatExecutionCount());
+        assertEquals(50L, stats.getTotalTimeBetweenExecutions());
+        assertEquals(50L, stats.getAvgTimeBetweenExecutions());
+        assertEquals(50L, stats.getMinTimeBetweenExecutions());
+        assertEquals(50L, stats.getMaxTimeBetweenExecutions());
+    }
+
+    public void testRepeatedRunStats() {
+        EnrichPolicyExecutionStatsTracker tracker = new EnrichPolicyExecutionStatsTracker();
+        tracker.captureCompletedExecution("test", 1000, 1050);
+        tracker.captureCompletedExecution("test", 1075, 1125);
+        tracker.captureCompletedExecution("test", 1200, 1250);
+        tracker.captureCompletedExecution("test", 1300, 1350);
+        tracker.captureCompletedExecution("test", 1400, 1450);
+        EnrichStatsAction.Response.ExecutionStats stats = tracker.getStats();
+        assertNotNull(stats);
+        assertEquals(5L, stats.getTotalExecutionCount());
+        assertEquals(250L, stats.getTotalExecutionTime());
+        assertEquals(25L, stats.getMinExecutionTime());
```

Comment:
Shouldn't both line 91 and 92 be `50L` instead of `25L` and `75L` respectively?

Reply:
This and many more test issues! I hope I got them all this time around